### PR TITLE
Change/merchant reference eng 189

### PR DESCRIPTION
**Pull request by andivido**
- Replaces Order ID with merchant_reference in the metadata
- Fixes an reincorporates the use of a shared secret authentication
- Aligns webhook platform information with application metadata
- Removes personal information from debugger

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -17,8 +17,8 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
     const CALLBACK_PATH      = 'rest/V1/divido/update/';
     const REDIRECT_PATH      = 'divido/financing/success/';
     const CHECKOUT_PATH      = 'checkout/';
-    const VERSION            = '2.1.0';
-    const WIDGET_LANGUAGES   = ["en", "fi" , "no", "es", "da", "fr", "de"];
+    const VERSION            = '2.3.0';
+    const WIDGET_LANGUAGES   = ["en", "fi" , "no", "es", "da", "fr", "de", "se", "pe"];
```
Comment (freddieline) [inline Helper/Data.php:21]: swedish should be "sv" and are you sure "pe" is just spanish. It doesn't exist in https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes

Comment (andivido) [reply Helper/Data.php]: Cheers, I've removed se as guess we're not supporting Uralic (se) any time soon! sv seems to hide my widget, hence the need for a list. pe is a weird one, as the widget is geared to receive this [https://github.com/dividohq/calculator-web-pub/blob/63187c6e5d7b08a9880bb021e08c98ab5f5a5d64/components/widget/index.tsx - line 40]

**Comment by freddieline** [comment]
The version needs to be updated in 
https://github.com/dividohq/finance-plugin-magento2/blob/master/etc/module.xml
Like so
https://github.com/dividohq/finance-plugin-magento2/pull/69

I think this pushes to composer

File: Helper/Data.php [hunk 2]
```diff
@@ -460,7 +459,7 @@ public function creditRequest($planId, $depositAmount, $email, $quoteId = null)
             ->withMetadata(
                 [
                     'initial_cart_value'    => $grandTotal,
-                    'quote_id'              => $quoteId,
+                    'merchant_reference'    => $quoteId,
```
[the remark this answers is not in the document]
Comment (andivido) [reply Helper/Data.php]: This was resolved in c94b3cb

**Pull request merged**